More tests

notCalle opened the pull request:
Expanded test coverage for `AttributeNode` behaviour of `Vertex` and `Edge`.

notCalle committed:
Proper enumeration expectation phrasing

Use the expect(enum).to have(count).items descriptive style
instead of explicit size comparison.

notCalle committed:
Common examples for all kinds of attribute nodes

notCalle committed:
Include attrnode examples in Vertex specs

The examples for all kinds of `KVDAG::AttributeNode` are kept in their
own set of shared examples. Include them in the specs for `KVDAG::Vertex`.

notCalle committed:
Add specs for KVDAG::Edge

Included shared examples for attrnode.

creideiki reviewed (changes requested):
This doesn't work, because `have` is undefined:

```
  3) KVDAG with first vertex has one vertex
     Failure/Error: expect(@dag.vertices).to have(1).item

     NoMethodError:
       undefined method `have' for #<RSpec::ExampleGroups::KVDAG::WithFirstVertex:0x000000023a9268>
     # ./spec/kvdag_spec.rb:35:in `block (3 levels) in <top (required)>'
```

You need to depend on `rspec/collection_matchers`. See e.g. saab-simc-admin/palletjack@a3fdcff6 and saab-simc-admin/palletjack@da31a2f1.

notCalle committed:
Update rspec dependencies to match Pallet Jack

To ease development of related components, ensure that dependencies
are kept in sync.
